---
title: "Final Project"
pagetitle: "BIOL 3100"
author: "Kate Hickman"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 4
    toc_float: yes
TOC: TRUE
---

## [Home](https://kjehickman.github.io/) | [Final Project](https://kjehickman.github.io/Final_Project/Final_Project.html) | [About Me](https://kjehickman.github.io/About_Me/About_me.html)

```{r optionsetup, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE)
```
<br>

# <b>Bacterial-Fungal Interactions in Mangrove Roots</b>

## Introduction

<br>
This analysis aims to illuminate potential interactions between bacterial and fungal species in terms of community assembly as a function of location, host plant or plant structure. The samples for this analysis were taken from mangroves trees in Southeast Asia and have been sequenced, targeting the 16S and ITS1 genes for bacteria and fungi, respectively. 
<br><br>


```{r setup, include=FALSE}
library(tidyverse); packageVersion("tidyverse")
library(phyloseq); packageVersion("phyloseq")
library(ShortRead); packageVersion("ShortRead")
library(ggplot2)
library(geosphere)
library(vegan)
library(microbiome)
library(BiocManager)
library(devtools) 
library(patchwork)

plot_bar2 <- function (physeq, x = "Sample", y = "Abundance", fill = NULL, 
                       title = NULL, facet_grid = NULL) 
{
  mdf = psmelt(physeq)
  p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
  p = p + geom_bar(stat = "identity", position = "stack")
  p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0))
  if (!is.null(facet_grid)) {
    p <- p + facet_grid(facet_grid)
  }
  if (!is.null(title)) {
    p <- p + ggtitle(title)
  }
  return(p)
}
```

# <b>Data Analysis</b>


## Bacterial and Fungal Community Data

The data in question is comprised of several data sets, resulting from sequencing published by Lee et a. (2019 and 2020) and is forthcoming in Wainright et al. These data illustrate the presence of microbial communities on Mangrove roots. Microbial communities of interest are **Bacteria** and **Fungi**--we will not consider **Archaea** since they are beyond the scope of this research. 

<br>

The data sets have been condensed into a "phyloseq" object, which is generally comprised of sample data, observational taxonomic unit (OTU) tables, taxonomy tables, and phylogenetic trees (not present in our data). The sample data contains a list of variables that present themselves as interesting predictors of community assembly--we will use microbe type and taxa as a proxy for community assembly. Below you will see the parameters for samples taken from two hosts at seven different sites across Southeast Asia.
<br>

```{r data load, echo=TRUE, include=TRUE, eval=TRUE}
full <- readRDS("./bact_and_fungi_clean_ps_object")

full_sam <- full@sam_data %>% as("data.frame")

full_sam %>% 
  select(Location, Host, Structure, Microbe, Lat, Lon) %>% 
  kableExtra::kable() %>% 
  kableExtra::kable_classic() %>% 
  kableExtra::kable_styling() %>% 
  kableExtra::scroll_box(width = "700px", height = "300px")
```

<br>

Now that we have seen the observed parameters, can can begin to ask questions about the data. For the purposes of this analysis, we'll be focusing on Structure, Microbe (including taxonomy), Host, and Location.

<br>
Before diving into an analysis, it's important to understand the shortcomings in your data--otherwise errors can compound to create patterns that may skew your results. 

<br>

```{r data discrepancies, echo=TRUE}
# full_sam %>% 
#   select(Host, Location, Structure) %>% 
#   filter(Location != "Blank") %>% 
#   filter(Host != "Blank") %>% 
#   table() %>% 
#   as.data.frame()

full_sam %>% 
  select(Host, Location, Structure) %>% 
  filter(Location != "Blank") %>% 
  filter(Host != "Blank") %>% 
  table() %>% 
  as.data.frame() %>% 
  ggplot(aes(x=Location, y=Freq, color=Host)) +
  geom_point() +
  facet_wrap(~Structure) +
  labs(y = "Number of Successful Samples") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))

```

<br><br>
In our data, there were likely several errors in gathering and storing the samples, as well as in the sequencing process and bioinformatic analysis. This has led to a few inconsistencies in the number of successful samples, as displayed in the graph above. 

<br>
For example, a direct comparison of samples between hosts in some locations, such as Merang, Redang or Tioman, would not yield useful results since some hosts have 10 samples while others have none (the species with no samples may have been absent from the sampling location, which has its own implications). Acknowledging these trends in data collection and analysis are crucial to the publication and dissemination of a successful, peer-reviewed study.

<br>
It may be useful to know which taxa were present in these samples, so here is the taxonomic data retrieved:
<br>

```{r tax_table, echo=FALSE}
full_tax <- as.data.frame(full@tax_table)

full_tax %>% 
  kableExtra::kable() %>% 
  kableExtra::kable_classic() %>% 
  kableExtra::kable_styling() %>% 
  kableExtra::scroll_box(width = "700px", height = "300px")

```

<br>

# <b>Analysis 1: Factors Impacting Diversity & Distribution</b>

<br>


## Multiple Regression Model: Location v. OTU

<br>

To begin the analysis, I wanted to start out broad and then narrow down the factors impacting community assembly. This led me to my first question, "How does geographic location impact the species present?" To answer this question, I decided to run a few models. 
<br><br>
Statistical analysis can be a useful way of validating (or refuting) patterns we think we see in the data. I used a generalized linear model incorporating the latitude and longitude of a sample as predictors of microbe type (remember, our only options are Fungi and Bacteria).
<br>
```{r glm bact, echo=FALSE}

samp <- data.frame(sample_data(full))

samp <- samp %>% 
  mutate(Bact = case_when(Microbe == "Bacteria" ~ TRUE,
                          TRUE ~ FALSE)) %>% 
   mutate(Bact = as.logical(Bact))

glm(formula = Bact ~ Lon + Lat, data = samp, family="binomial") %>% summary()

```
<br>
Based on the intercept and positive values for Lon/Lat, the model predicts that as you move northeast the chances of a given microbe being bacterial increases.
<br><br>
To see if this tracked across samples in a broader trend, I used a Multiple Regression Model (MRM) on distance matrices. The MRM is designed to compare the similarity of two samples on a scale of 0 to 1, where 0 means the two samples are exactly the same and 1 means there is no similarity between them. 
<br><br>
I began by making a matrix containing only the geographical locations (latitude and longitude) of each sample. Then I created another matrix with the operational taxonomic units to gauge the diversity of each sample. 
<br><br>
Once I made those matrices, I regressed the matrices against each other using the response matrix (in this case, otu_dist) as a function of the explanatory matrix (location_dist).
<br>

```{r martices, echo=FALSE}
lat <- full@sam_data$Lat %>% as.numeric()
lon <- full@sam_data$Lon %>% as.numeric()

location_dist <- data.frame(lat, lon) %>% dist()

otu_dist <- otu_table(full) %>% dist()

otu_dist <- vegan::vegdist(otu_table(full), na.rm = TRUE)
```

To illustrate the similarity of each of the samples, a heatmap can act as a useful tool. The otu_dist matrix as a heat map displays each samples' similarity to the other samples in terms of the community composition.
<br><br>
```{r heatmap, echo=FALSE}
otu_dist %>% as.matrix() %>% heatmap()

```

<br>
Here are the results of the regression of otu_dist against location_dist:
<br>

```{r MRM1, echo=TRUE}
MRM1 <- ecodist::MRM(otu_dist ~ location_dist)

MRM1
```
<br>
While the p-value is statistically significant, the correlation between diversity and proximity is rather weak. According to the R2 value, geographic location can only explain about 1% of the variation in sample diversity. 
<br><br>
Since the broad picture isn't specific enough to explain detailed trends, I decided to move down a level. 
<br>


# <b>Analysis 2</b>

<br>


## Diversity of Phyla by Structure


<br><br>
Rather than predict diversity by geographical proximity, I'm interested to know if the structure of the host plant may play a more influential role in microbial community diversity. These structures, including Fruit, Leaf, Pneumatophore, and Sediment, create distinct environmental conditions that allow microbes to specialize. Scaled to size, they are they represent the microclimates for the microorganisms and, thus, the next variable to analyze. 
<br><br>
After removing the "Blanks" from the sample data for Structure, I created a stacked bar graph that displays various phyla and their relative abundance by structure. 
<br>

```{r phyla ~ structure, echo=TRUE, fig.height=7, fig.width=9}
full_structures <- full %>% subset_samples(Structure != "Blank")

# this shows phylum abundance by structure
merge_samples(full_structures, "Structure") %>% 
  transform_sample_counts(fun = function(x){x/sum(x)}) %>% 
  plot_bar(fill = "Phylum")
```
<br>
Now that we can visualize those relationships, how accurate are our perceptions of correlation? In other words, do the patterns we project correlate to reality? Let's use a generalized linear model to find out. 
<br>
```{r glm structure, echo=FALSE, include=TRUE}
samp2 <- data.frame(sample_data(full_structures))

samp2 <- samp2 %>% 
  mutate(Bact = case_when(Microbe == "Bacteria" ~ TRUE,
                          TRUE ~ FALSE)) %>% 
   mutate(Bact = as.logical(Bact))

glm(formula = Bact ~ Host + Structure, data = samp2, family = binomial) %>% summary()
```

<br><br>
There are a few Archaea samples in our data and I'm going to remove those, as they're beyond the scope of this research. Then we can show a plot that gives relative abundance by Kingdom. 

```{r nonbact parameters, echo=TRUE}
full_nonarch <- subset_taxa(full, Kingdom != "Archaea")

# quick plot to look at kingdom-level taxonomy
full_nonarch %>% 
  subset_samples(Location != "Blank") %>% 
  merge_taxa("Genus") %>% 
  merge_samples("Location") %>% 
  transform_sample_counts(function(x){x/sum(x)}) %>%
  plot_bar2(fill="Kingdom")

```

SO.. HOW DOES STRUCTURE AFFECT DIVERSITY?? ... AND WHY DOESN'T THIS GRAPH WORK?
  REMOVE BLANK

<br><br>

## Relative Class Abundance by Location

Now that we've taken a look at structure, we can break down the Phyla distribution by host. We had two types of hosts in the data-collection: *Avicennia alba* and *Sonneratia alba*. 
<br><br>
The first graph shows the phyla distribution as differentiated by host. However, the downside to this graph is that the Phyla are sorted alphabetically rather than by microbe-type. 
<br><br>
The second graph has undergone some modification, based off a few suspicions: 
<br><br>
  1. I suspect both the host and the structure impact the diversity of the microbial community
<br><br>
2. The impact of those variables will likely be different for Fungi vs. Bacteria<br>
<br><br>
Thus, the second graph shows a separate column for the outcome of the three variables interacting. The first 8 rows go through the structures of *Avicennia alba* by microbe, and the second 8 do the same thing for *Sonneratia alba*. 
<br><br>

```{r largefig, echo=FALSE}

# can add these fig dimensions later: fig.align='center', echo=FALSE, fig.height= "3", fig.width= "3"
full_nonarch %>% 
  subset_samples(Host != "Blank") %>% 
  merge_samples("Host") %>% 
  transform_sample_counts(function(x){x/sum(x)}) %>% 
  plot_bar(fill="Phylum") +
  facet_wrap(~"Microbe")

newmerge <- paste0(full@sam_data$Host, "_", full@sam_data$Structure)

full@sam_data$newmerge <- newmerge

full %>% 
  subset_samples(Host != "Blank") %>% 
  merge_samples("newmerge") %>% 
  transform_sample_counts(function(x){x/sum(x)}) %>% 
  plot_bar(fill="Phylum") +
  scale_fill_viridis_d()
```
<br><br><br><br>
LET'S MAKE THIS GRAPH LARGER: in code chunk, 
<br><br>
Alright, so that graph is pretty hard to read. Merging those variables made the data a little messy, so it needs to be restored. Once that is accomplished, we can split the previous graph up into several graphs that make the data a bit more digestible. 

```{r fix data, echo=TRUE}
# reassign the earlier plot
# full2 <- full %>% 
#   subset_samples(Host != "Blank") %>% 
#   merge_samples("newmerge") %>% 
#   transform_sample_counts(function(x){x/sum(x)}) %>% 
#   plot_bar(fill="Phylum") +
#   scale_fill_viridis_d()

# this will fix the otu_table slot
full2 <- full %>% 
  subset_samples(Host != "Blank") %>% 
  merge_samples("newmerge") 

# this code repairs the variables that were damaged upon merging
full2@sam_data$Host <- row.names(full2@sam_data) %>% str_split("_") %>% map_chr(1)
full2@sam_data$Structure <- row.names(full2@sam_data) %>% str_split("_") %>% map_chr(2)
#full2@sam_data$Microbe <- row.names(full2@sam_data) %>% str_split("_") %>% map_chr(3)

# this graph shows separate columns for each host, and is wrapped by structure
full2 %>% 
  transform_sample_counts(function(x){x/sum(x)}) %>% 
  plot_bar2(x="Host", fill="Phylum")+
  scale_fill_viridis_d()+
  facet_wrap(~Structure)
#added linw 248
```

CONSIDER EDITING OUT THE LARGE GRAPHS
<br><br><br> Now that we can see the diversity of phyla split by host and structure, we can further our analysis to include the microbes. 
<br><br>To show those same differences by microbe, we'll make two graphs: one for bacteria, and another for fungi.

```{r bact_plt, echo=TRUE}

# here is our bacterial plot
bact_plot <- full2 %>% 
  subset_taxa(Kingdom =="Bacteria") %>% 
  transform_sample_counts(function(x){x/sum(x)}) %>% 
  plot_bar2(x="Host", fill="Phylum")+
  scale_fill_viridis_d()+
  facet_wrap(~Structure) +
  labs(title = "Bacterial Community by Host Structure") +
  theme(plot.title = element_text(size = 10, face = "bold"), 
        plot.margin = margin(t = 1, r = 1, b = 1, l = 1, unit = "cm"))

bact_plot

# this will combine the plots to show phyla sorted by kingdom
#bact_plot / fung_plot
# REMOVE THIS--ITS TOO MUCH
```

HELP WITH AXIS TITLES

<br><br>
This plot shows fungal community assembly, differentiated by the host plant's sturcture. The most uniform communtiy appears to exist on the fruit, whereas the most speciose community exists in the sediment. This trend of diverse community in sediment is observed in bacterial communities as well.
<br><br>
Both bacterial and fungal communities appear to be more diverse in submerged environments, such as the pneumatophores and sediment, as compared to terrestrial structures. This finding is rather ironic, considering that much more is known, generally speaking, about terrestrial fungi than marine fungi. The number of "unidentified" species in these samples is liekly quite high.
<br><br>

```{r fung plat, echo=TRUE}

# here is our fungal plot
fung_plot <- full2 %>% 
  subset_taxa(Kingdom =="Fungi") %>% 
  transform_sample_counts(function(x){x/sum(x)}) %>% 
  plot_bar2(x="Host", fill="Phylum")+
  scale_fill_viridis_d()+
  facet_wrap(~Structure) +
  labs(title = "Fungal Community by Host Structure")
#  theme(title = "Fungal Community by Host Structure")

fung_plot
```


THESE AXES NEED TO BE MESSED WITH: make the y-axes say fungal/bacterial abundance, fix the legend
*** ALSO NEED TO TRANSFORM TO RELATIVE ABUDANCE AND MAKE Y-AXIS 0-1
<br><br>

# <b>Analysis 3</b>

<br><br>
Now that we've demonstrated that diversity exists, and is impacted by several different variables, it would be prudent to know how many different alpha-diversity metrics we're working with. The estimate_richness() and plot_richness() functions are helpful in this analysis. This will help us to understand the ecological structure of the community in terms of richness and / or evenness. 
<br><br><br>

## Alpha Diversity and Species Richness

<br><br>

```{r primary richness, echo=TRUE}
# this pruning prepares the data for richness
GP <- prune_species(speciesSums(full) > 0, full)

# this graph wraps by diversity type (Chao1, Shannon) and separates based on structure
GP %>% 
  subset_samples(Host != "Blank") %>%
  plot_richness(x = "Structure", 
              measures=c("Observed", "Shannon"),
              sortby = "Observed")
```


This graph illustrates the greater number of diversity measures in sediment, reflecting the information provided in "WHICH OTHER PLOTS SHOW HIGHER DIVERSITY IN SEDIMENT?"
<br><br><br>
Now we can analyze richness through the distribution of microbes (graph 1), and also by structure (graph 2).

```{r richness by color, echo=TRUE}
# colored by kingdom/microbe 
plot_richness(full, measures = c("Observed","Shannon"), 
              color = "Microbe", 
              sortby = "Observed") +
  theme(axis.text.x = element_blank())

# get code with plot.richness and in gemo_point section of code, add metric for alpha so that points can be read more easily

alpha <- estimate_richness(full, measures = c("Observed", "Shannon"))
alpha$SampleID <- row.names(alpha)

full_sam2 <- full_join(full_sam, alpha, by="SampleID")

full_sam2 %>% names
full_sam2 %>%
  arrange(Observed) %>% 
  mutate(SampleID=factor(SampleID, levels=SampleID)) %>% 
  ggplot(aes(x=SampleID, y=Observed, color=Structure)) +
  geom_point() +
  theme_minimal() +
  theme(axis.text.x = element_blank())

# colored by structure
plot_richness(full, 
              measures = c("Observed","Shannon"), 
              color = "Structure", 
              sortby = "Observed") +
  theme(axis.text.x = element_blank())
```

These graphs demonstrate the greater speciosity and diversity of fungal species, as well as the richness in sediments. 
HELP WITH HOW TO READ THEM EXACTLY, CHECK OUT X-AXIS AND MAKE POINTS MORE READABLE BY WRIT
DOES SEDIMENT MATTER FOR WHICH SPECIES THE PLANT NEEDS - ARE MICROBES A SUBSET OF SEDIMENT WHERE THEY'RE FOUND AND HOW WOULD THAT IMPACT NURSERIES
<br><br>
Grouping by the microbe and utilizing a boxplot clarifies the data visually. 

```{r rich boxplot, echo=TRUE}
plot_richness(full, 
              x = "Microbe",
              measures = c("Observed","Shannon"), 
              color = "Structure", 
              sortby = "Observed") +
  geom_boxplot(alpha = .5) +
  theme_minimal()
```


# <b>Conclusion</b>

3 BULLET POINTS

NAME ANALYSES (NOT NUMBERS)




